### ISSUE TITLE
Support For Bitbucket [Provider]

### ISSUE THREAD

**Issue by babatundeakinse** [adamwathan/eloquent-oauth#47]
Hi Adman, Is can you add bit-bucket to the list of providers ? thanks :)

**Comment by adamwathan** [comment]
Definitely, I'm working on a big refactor at the moment that pulls all of that code out of this package, so I'll add BitBucket when that's done.

If you're interested here's a preview:

http://github.com/adamwathan/socialnorm
http://github.com/adamwathan/socialnorm-github
http://github.com/adamwathan/socialnorm-facebook
http://github.com/adamwathan/socialnorm-google
http://github.com/adamwathan/socialnorm-linkedin
http://github.com/adamwathan/socialnorm-soundcloud
http://github.com/adamwathan/socialnorm-instagram

You can see how it's going to be integrated by taking a look at the `aw-socialnorm` branch on this repo :)

**Comment by babatundeakinse** [comment]
Thanks Adam, will be watching this closely :+1:

**Comment by drbyte** [comment]
Bitbucket now appears to offer OAUTH2: 
- https://developer.atlassian.com/bitbucket/concepts/oauth2.html
- https://confluence.atlassian.com/bitbucket/users-endpoint-423626336.html

One of the appeals of BitBucket was the unlimited private repos ... but of course today GitHub now offers unlimited repos for one base price: https://github.com/blog/2164-unlimited-private-repos

Admittedly there are other appeals to BitBucket especially when invested in other Atlassian products ;)
